mail_tracking/models/mail_tracking_email.py

```diff
@@ -0,0 +1,322 @@
+    recipient_address = fields.Char(
+        string='Recipient email address', readonly=True, store=True,
+        compute='_compute_recipient_address', index=True)
+    sender = fields.Char(string='Sender email', readonly=True)
+    state = fields.Selection([
+        ('error', 'Error'),
+        ('deferred', 'Deferred'),
+        ('sent', 'Sent'),
+        ('delivered', 'Delivered'),
+        ('opened', 'Opened'),
+        ('rejected', 'Rejected'),
+        ('spam', 'Spam'),
+        ('unsub', 'Unsubscribed'),
+        ('bounced', 'Bounced'),
+        ('soft-bounced', 'Soft bounced'),
+    ], string='State', index=True, readonly=True, default=False,
+        help=" * The 'Error' status indicates that there was an error "
+             "when trying to sent the email, for example, "
+             "'No valid recipient'\n"
+             " * The 'Sent' status indicates that message was succesfully "
+             "sent via outgoing email server (SMTP).\n"
+             " * The 'Delivered' status indicates that message was "
+             "succesfully delivered to recipient Mail Exchange (MX) server.\n"
+             " * The 'Opened' status indicates that message was opened or "
+             "clicked by recipient.\n"
+             " * The 'Rejected' status indicates that recipient email "
+             "address is blacklisted by outgoing email server (SMTP). "
+             "It is recomended to delete this email address.\n"
+             " * The 'Spam' status indicates that outgoing email "
+             "server (SMTP) consider this message as spam.\n"
+             " * The 'Unsubscribed' status indicates that recipient has "
+             "requested to be unsubscribed from this message.\n"
+             " * The 'Bounced' status indicates that message was bounced "
+             "by recipient Mail Exchange (MX) server.\n"
+             " * The 'Soft bounced' status indicates that message was soft "
+             "bounced by recipient Mail Exchange (MX) server.\n")
+    error_smtp_server = fields.Char(string='Error SMTP server', readonly=True)
+    error_type = fields.Char(string='Error type', readonly=True)
+    error_description = fields.Char(
+        string='Error description', readonly=True)
+    bounce_type = fields.Char(string='Bounce type', readonly=True)
+    bounce_description = fields.Char(
+        string='Bounce description', readonly=True)
+    tracking_event_ids = fields.One2many(
+        string="Tracking events", comodel_name='mail.tracking.event',
+        inverse_name='tracking_email_id', readonly=True)
+
+    @api.model
+    def _email_score_tracking_filter(self, domain, order='time desc',
+                                     limit=10):
+        """Default tracking search. Ready to be inherited."""
+        return self.search(domain, limit=limit, order=order)
+
+    @api.model
+    def email_is_bounced(self, email):
+        if email:
+            return len(self._email_score_tracking_filter([
```

Comment:
wouldn't be better to use search_count here, especially if - as stated above - we gonna have tons of records?